src/contracts/interfaces/IAllocationManager.sol

```diff
@@ -0,0 +1,331 @@
+    /// @dev Thrown when an operator attempts to set their allocation delay to 0
+    error InvalidDelay();
+    /// @dev Thrown when an operator's allocation delay has yet to be set.
+    error UninitializedAllocationDelay();
+    /// @dev Thrown when provided `expectedTotalMagnitude` for a given allocation does not match `currentTotalMagnitude`.
+    error InvalidExpectedTotalMagnitude();
+    /// @dev Thrown when an invalid operator set is provided.
+    error InvalidOperatorSet();
+    /// @dev Thrown when an invalid operator is provided.
+    error InvalidOperator();
+    /// @dev Thrown when caller is not the delegation manager.
+    error OnlyDelegationManager();
+    /// @dev Thrown when an operator attempts to set their allocation for an operatorSet to the same value
+    error SameMagnitude();
+    /// @dev Thrown when an allocation is attempted for a given operator when they have pending allocations or deallocations.
+    error ModificationAlreadyPending();
+    /// @dev Thrown when an allocation is attempted that exceeds a given operators total allocatable magnitude.
+    error InsufficientAllocatableMagnitude();
+    /// @dev Thrown when attempting to use an expired eip-712 signature.
+    error SignatureExpired();
+    /// @dev Thrown when attempting to spend a spent eip-712 salt.
+    error SaltSpent();
+
+    /**
+     * @notice struct used to modify the allocation of slashable magnitude to list of operatorSets
+     * @param strategy the strategy to allocate magnitude for
+     * @param expectedTotalMagnitude the expected total magnitude of the operator used to combat against race conditions with slashing
+     * @param operatorSets the operatorSets to allocate magnitude for
+     * @param magnitudes the magnitudes to allocate for each operatorSet
+     */
+    struct MagnitudeAllocation {
+        IStrategy strategy;
+        uint64 expectedTotalMagnitude;
+        OperatorSet[] operatorSets;
+        uint64[] magnitudes;
+    }
+
+    /**
+     * @notice struct used for pending free magnitude. Stored in (operator, strategy, operatorSet) mapping
+     * to be used in completeDeallocations.
+     * @param magnitudeDiff the amount of magnitude to deallocate
+     * @param completableTimestamp the timestamp at which the deallocation can be completed, 21 days from when queued
+     */
+    // struct PendingFreeMagnitude {
+    //     uint64 magnitudeDiff;
+    //     uint32 completableTimestamp;
+    // }
+
+    /**
+     * @notice struct used for operator magnitude updates. Stored in _operatorMagnitudeInfo mapping
+     * @param currentMagnitude the current magnitude of the operator
+     * @param pendingMagnitudeIdff the pending magnitude difference of the operator
+     * @param effectTimestamp the timestamp at which the pending magnitude will take effect
+     */
+    struct MagnitudeInfo {
+        int128 pendingMagnitudeDiff;
+        uint64 currentMagnitude;
+        uint32 effectTimestamp;
+    }
```

Comment:
Not a fan of ordering the fields like this -- IMO unless absolutely needed for storage packing, fields should be logically grouped by function. So, `currentMagnitude` should not sit between a pending diff and its effective timestamp.